manual_run/weight_run.py

```diff
@@ -0,0 +1,118 @@
+from manual_run.default_scenario import DefaultScenario
+from manual_run.map_columns import MapColumns
+from manual_run.variables import Variables
+
+
+class WeightRun(Run):
+    def start(self, num_trials: int = 1, generate_graphs: bool = True):
+        scenario = DefaultScenario()
+
+        # Metrics to be used for evaluation
+        metrics = [
+            AverageCosineDifference(
+                name="Score Cosine Difference",
+                attribute_filter=[Attributes.SCORE.value],
+            ),
+            AverageCosineDifference(
+                name="Timeslot Cosine Difference",
+                attribute_filter=[Attributes.TIMESLOT_AVAILABILITY.value],
+            ),
+        ]
+        student_provider = DataProvider()
+        artifact: SimulationSetArtifact = SimulationSet(
+            settings=SimulationSettings(
+                num_teams=ceil(student_provider.num_students / Variables.team_size),
+                scenario=scenario,
+                student_provider=student_provider,
+            ),
+            algorithm_set={
+                AlgorithmType.PRIORITY: [
+                    PriorityAlgorithmConfig(
+                        MAX_KEEP=15,  # max num of solutions the algorithm is allowed to keep
+                        MAX_SPREAD=30,  # max num of team sets genereated by mutation at each step
+                        MAX_ITERATE=30,  # max num of iterations the algorithm is allowed to run
+                        MAX_TIME=100000,  # max time in seconds the algorithm can run for.
+                    ),
+                ]
+            },
+        ).run(num_runs=1)
+        team_set = list(artifact.values())[0][0][0]
+
+        self.create_csv(team_set, student_provider)
+
+    def create_csv(self, team_set, student_provider):
+        Variables.data_fields[0].append("TeamSizeViolation")
+        Variables.data_fields[0].append("TeamId")
+
+        for team in team_set.teams:
+            for student in team.students:
+                attributes = student.attributes
+
+                unique_id = student_provider.get_student(student.id)
+                time_slot = attributes[Attributes.TIMESLOT_AVAILABILITY.value][0]
+                tutor_preference = attributes[Attributes.TUTOR_PREFERENCE.value][0]
+                group_size = attributes[Attributes.GROUP_SIZE.value][0]
+
+                data_fields_input_1 = MapColumns.revert_timeslot(time_slot)
+                data_fields_input_2 = MapColumns.revert_tutor_preference(
+                    tutor_preference
+                )
+                data_fields_input_3 = MapColumns.revert_group_size(group_size)
+
+                # Process score and team size violation
+                positive_z = "1" if attributes[Attributes.SCORE.value][0] == 1 else "0"
+                team_size = len(team.students)
+
+                # Can be customize to fit the team size preference
+                team_size_violation = (
```

Comment:
you're right I dislike this, kinda care a lot less above this create_csv function and can learn to accept it if we're just choosing to do something greedy in the returned CSV. We should ponder how needed it is though

Reply:
I don't personally see much of a use case for this **team_size_violation** attribute as it is very specific to each situation. I assuming that you're kinda thinking the same thing here. Would be this something I could remove?

Reply:
I get the vibe Bowen probably wants it still so I'd just leave it in, maybe toss a comment so someone in the future knows they can remove it without breaking anything